File: wk6/credit.py
class Credit():
    #MASTERCARDS 51, 52, 53, 54, or 55
    #AMEX 34 or 37

    card_number = input("Enter your credit card number: ")
    mastercards = [51, 52, 53, 54, 55]
    amexes = [34, 37]
    visas = [4]

    sum = 0
    digits = 0

    try:
        for i in range(len(card_number) - 1, 0):#every other *2 starting w/ second to last digit
            if digits % 2 == 0:#if it is even
                value = 2 * int(card_number[i])
            else:
                value = int(card_number[i])
            digits += 1
            sum += value
        if sum % 10 == 0:
            prefix = int(card_number[0:2])
            if(int(card_number[0]) == 4):
                print("Visa")
            elif prefix in mastercards:
                print("Mastercard")
            elif prefix in amexes:
                print("Amex")
            else:
                print("Card number not visa or mastercard or amex")
        else:
            print("Card number invalid. Must fit algorithm")

    except ValueError:
        print("Card number invalid. Must be integer")

File: wk1/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void){
    char buffer[100];
    long number = -1;
    int sum = 0;
    while(number < 0){
        printf("Enter your card number:");
        if(fgets(buffer, sizeof(buffer), stdin)){
            if(1 == sscanf(buffer, "%ld", &number)){
                printf("%ld\n", number);
            }
        }
    }

    //multiply every other number starting at second to last
    for(long i = number/10; i > 0; i/=100){
        int digit = (2 * (i%10));
        if(digit >= 10){
            sum+= digit%10;//ones place
            sum+= digit/10;//tens place
        }
        else{
            sum += digit; //doubles then adds the digits of the product
        }
    }
    //add not multiplied digits
    for(long i = number; i > 0; i/=100){
        sum += i%10;
    }
    //% by 10 == 0 then all is good
    if(sum % 10 == 0){
        printf("VISA");
    }else{
        printf("%i\n", sum);
        printf("Invalid Number\n");
    }
    //for some reason the sum is 19 instead of 20 for the 4003600000000014 card
}
